spec/features/questionings_form_spec.rb

```diff
@@ -87,6 +87,15 @@
         visit edit_questioning_path(qing, locale: "en", mode: "m", mission_name: get_mission.compact_name)
         expect(page).to have_content("powerup.mp3")
       end
+
+      context "errors" do
+        it "should render error properly when there is no type" do
+          visit("/en/m/#{get_mission.compact_name}/forms/#{form.id}/choose-questions")
+          fill_in "Code", with: "foodie"
+          click_on("Save")
+          expect(page).to have_content("Question is invalid")
+        end
+      end
```

Comment:
Nice job on this spec. It is correct and covers the issue appropriately.

However I think it should rather be a model spec. Remember the testing pyramid. We try to save feature specs for integrations that model specs can't cover. But you can cover this case in the DynamicPatternValidator model spec, no?